### ISSUE TITLE
Problem with API paramter serializer in scala 3 [boopickle].

### ISSUE THREAD

**Issue by bjekt** [cornerman/sloth#203]
Hi 
Cool library! 

I'm struggling to get this working 100% for scala3: When using the Api specified in the docs:
```
trait Api:
  def fun(s: Int): Future[Int]
```
```
[error] -- [E172] Type Error: /**/bumper/Test.scala:71:33 
[error] 71 |  val api: Api = client.wire[Api]
[error]    |                                 ^
[error]    |No given instance of type chameleon.Serializer[(s : Int), java.nio.ByteBuffer] was found for parameter serializer of method execute in class ClientImpl.
[error]    |I found:
[error]    |
[error]    |    chameleon.ext.boopickle.boopickleSerializerDeserializer[(s : Int)](
[error]    |      <empty> :boopickle².Pickler[(s : Int)]:boopickle².Pickler[(s : Int)])
[error]    |
[error]    |But method generatePickler in trait MaterializePicklerFallback does not match type boopickle².Pickler[(s : Int)]
[error]    |
[error]    |where:    boopickle  is a object in package chameleon.ext
[error]    |          boopickle² is a package
[error]    |.

```

If changing to no params in fun it compiles and works perfectly. 
```
trait Api:
  def fun(): Future[Int]
```

**Comment by bjekt** [comment]
It look like this is related to when the parameter type is a primary type (Int, String etc). If a case class it works well as MaterializePicklerFallback (in boopickle) is able to handle it somehow. I'm not sure what this type parameter is `[(s : Int)]`. I guess if it would be Int instead there would be no problem

**Comment by bjekt** [comment]
A reproduction of the bug can be found here: https://github.com/cornerman/sloth/compare/master...bjekt:sloth:param-test

Possible workaround is to wrap and primary type parameters in case classes.

**Comment by cornerman** [comment]
@bjekt Thank you for opening this issue and providing a reproduction <3

I will try to look into it somewhere today. Will keep you updated about the progress.

**Comment by cornerman** [comment]
For a workaround in the moment, another alternative might be to add a dummy parameter to single methods. I have not tested it, but I would guess, this could work:

```
def single(number: Int, dummy: Unit = ()): Future[Int] = ???
```

**Comment by cornerman** [comment]
Fix is released in version `0.7.1`.

I have added a test-case accordingly. Still, please let me know, if the issue persists for you.
